src/ts/greedy-nav/GreedyNav.ts

```diff
@@ -0,0 +1,1062 @@
+        );
+        this.toggleWrapper.classList.add('priority-nav__wrapper');
+
+        _this.classList.add('priority-nav');
+    }
+
+    /**
+     * Bind eventlisteners
+     */
+    listeners(_this: HTMLElement): void {
+        // Check if an item needs to move
+        // if (window.attachEvent) {
+        //     window.attachEvent('onresize', () => {
+        //         if (priorityNav.doesItFit) priorityNav.doesItFit(_this);
+        //     });
+        // } else if (window.addEventListener) {
+        window.addEventListener('resize', () => {
+            this.doesItFit(_this, this.settings.throttleDelay);
+        });
+
+        window.addEventListener(
+            'orientationchange',
+            () => {
+                this.doesItFit(_this, this.settings.throttleDelay);
+            },
+            true
+        );
+        // }
+        const {
+            navDropdownClassName,
+            navDropdownLabelActive,
+            navDropdownLabel
+        } = this.settings;
+
+        const navDropdownToggle = _this.querySelector<HTMLElement>(
+            this.navDropdownToggleSelector
+        );
+
+        if (navDropdownToggle) {
+            // Toggle dropdown
+            navDropdownToggle.addEventListener('mousedown', event => {
+                const navDropdown = _this.querySelector<HTMLElement>(
+                    this.navDropdownSelector
+                );
+
+                if (navDropdown) {
+                    event.stopPropagation();
+                    toggleClass(navDropdown, 'show');
+
+                    if (event.currentTarget) {
+                        toggleClass(
+                            <HTMLElement>event.currentTarget,
+                            'is-open'
+                        );
+                    }
+                    toggleClass(_this, 'is-open');
+
+                    /**
+                     * Toggle aria hidden for accessibility
+                     */
+                    if (_this.classList.contains('is-open')) {
+                        navDropdown.setAttribute('aria-hidden', 'true');
+
+                        updateLabel(
+                            _this,
+                            navDropdownLabelActive,
+                            this.navDropdownToggleSelector,
+                            this.settings.navDropdownLabelActive
+                        );
+
+                        navDropdown.blur();
+                    } else {
+                        navDropdown.setAttribute('aria-hidden', 'false');
+
+                        updateLabel(
+                            _this,
+                            navDropdownLabel,
+                            this.navDropdownToggleSelector,
+                            this.settings.navDropdownLabelActive
+                        );
+                    }
+                }
+            });
+        }
+
+        const lastItemCloseHandler = (event: FocusEvent) => {
+            if (this.toggleWrapper === null) {
```

Comment:
So here it's `===` and not `==` whereas above in all the cases, the wrapper toggle check was `==`. Should double check what is the intention.

Reply:
In this case it will either be null or a value, in other cases where `==` is used `undefined` is a posibility - I'll switch it to `!` in those cases as you seem to be more comfortable with type coercion to a boolean vs a null.

Reply:
Actually I'm gonna leave null in explicitly - the only two values it can be is HTMLElement or null so It doesn't matter if it's `==` or `===`. I'll put `===` as it'll raise less conversation.